File: android-working-old/app/src/main/java/uk/co/darkerwaters/scorepal/score/base/SimplePoint.java
package uk.co.darkerwaters.scorepal.score.base;

import android.content.Context;

import uk.co.darkerwaters.scorepal.R;

public class SimplePoint implements Point {
    private final int value;

    public SimplePoint(int value) {
        this.value = value;
    }

    @Override
    public int val() {
        return this.value;
    }

    @Override
    public String displayString(Context context) {
        return Integer.toString(this.value);
    }

    @Override
    public String speakString(Context context) {
        if (null != context && this.value == 0) {
            return context.getString(R.string.speak_zero);
        } else {
            return Integer.toString(this.value);
        }
    }

    @Override
    public String speakString(Context context, int number) {
        if (null != context && this.value == 0 && number != 1) {
            return context.getString(R.string.speak_zeros);
        }
        else {
            return speakString(context);
        }
    }

    @Override
    public String speakAllString(Context context) {
        return this.value + " " + context.getString(R.string.speak_all);
    }
}

File: multiphone/lib/match/points/simple_point.dart
import 'package:flutter/cupertino.dart';
import 'package:multiphone/helpers/values.dart';
import 'package:multiphone/match/points/point.dart';

class SimplePoint implements Point {
  final int value;

  SimplePoint(int value) : value = value;

  @override
  int val() {
    return this.value;
  }

  @override
  String displayString(BuildContext context) {
    return this.value.toString();
  }

  @override
  String speakString(BuildContext context) {
    if (null != context && this.value == 0) {
      return Values(context).strings.speak_zero;
    } else {
      return this.value.toString();
    }
  }

  @override
  String speakNumberString(BuildContext context, int number) {
    if (null != context && this.value == 0 && number != 1) {
      return Values(context).strings.speak_zeros;
    } else {
      return speakString(context);
    }
  }

  @override
  String speakAllString(BuildContext context) {
    return '$value ${Values(context).strings.speak_all}';
  }
}
